### ISSUE TITLE
How to control reference-image adherence in the edit region for Flux2-Klein? (edit region copies the reference)

### ISSUE THREAD

**Issue by houcem-fehri** [scraed/LanPaint#97]
**Environment**

- charrywhite/LanPaint-diffusers @ 2f96318, scraed/LanPaint @ d696d48, diffusers main
- Model: Flux2-Klein 9B, FluxKleinAdapter
- Use case: large-area reimagine, preserve a masked subject pixel-exact, regenerate a completely new background. Preserve fraction is small (~14% preserve / ~86% edit).

**What I'm trying to do**

Keep the masked subject locked, but have the edit region generate a genuinely new scene driven by the prompt.

LanPaintConfig(
    n_steps=2,            # Langevin "thinking" steps
    chara_lambda=8.0,
    cfg_big=5.0,          # Image First
    beta=1.0,
    step_size=0.3,
    friction=30.0,
    early_stop=1,
    blend_overlap=0,      # no dilating boundary blend (no hard paste-back either)
)

PROMPT = (
    "A photorealistic scene of an elegant balcony, where an ornate stone railing runs "
    "along the subject's lower edge with the subject standing just behind it. The balcony "
    "overlooks the French Riviera coastline, the deep blue Mediterranean Sea stretching to "
    "the horizon under a clear sky. Soft natural daylight, subtle shadows and highlights "
    "matching the existing subject's illumination, the same eye-level perspective, focal "
    "length and depth of field, so the result reads as one shot taken around the subject."
)
NEG = "duplicate person, second figure, cloned subject, extra limbs, blurry, seams, artifacts"

**What I observe**

The edit region adheres very strongly to the reference image and reproduces its structure (e.g. occluders/objects from the source "echo" into the new scene). In an iterative second pass (feed pass-1 output back in with the same mask), the edit region reproduces the input almost exactly (<1% pixel change) regardless of mask shape, mask size, or prompt.

**Root cause (Claude's reading of the code)**

FluxKleinAdapter.predict_x0 concatenates the full reference latent into every forward pass:
latent_model_input = torch.cat([x, self._y_packed], dim=1)
and CFG is computed with the reference present in both the conditional and unconditional passes, so guidance can't push away from it. The LanPaint preserve term (-(1+λ)(x_t − y) + λ(x_t − x0_big)) only applies to the preserve region (mask=1); the edit region is −(x_t − x0), but x0 still "sees" the concatenated reference.

**What I've tried (no effect on edit-region adherence)**

- cfg_big (5.0 → −1.0): inert — only scales one sub-term of the preserve-region score.
- chara_lambda: only affects the preserve region by construction.
- n_steps 2 → 5: no change.
- Prompt phrasing (descriptive vs. imperative "continue the scene"): no change.
- Second-pass refinement with ring / vertical-band masks of varying width: reproduces pass 1.

**What partially works**

Manually attenuating the reference latent in the edit region (scale _y_packed by α where mask=0):
- α=1.0 = current behavior (copies reference)
- α=0.0 (hard zero) = hallucinates / anatomy breaks (extra limbs)
- α≈0.3–0.7 = edit region departs from the reference and follows the prompt more, subject still preserved

**Questions**

1. Is there a supported/recommended way to control reference-image strength in the edit region specifically (an image-guidance / reference-CFG scale), separate from the text guidance_scale?
2. Is edit-region reference adherence expected to be this strong for Flux2-Klein, or am I missing a config?
3. Is reference attenuation (the α approach) reasonable, or would a true image-CFG formulation (an extra no-reference forward pass, extrapolating away from the reference) be the right fix? Any guidance on a stable range?
4. For large-area edits (preserve ≪ edit) on an edit-style model like Klein, is this simply the wrong tool, or is there a known config for it?

<img width="1080" height="1080" alt="Image" src="https://github.com/user-attachments/assets/8fcae52d-3d5c-4708-bc0a-1ebc2f984844" />

<img width="1080" height="1080" alt="Image" src="https://github.com/user-attachments/assets/aa460f42-a8f1-455d-a254-f0392c27a46f" />

<img width="1080" height="1080" alt="Image" src="https://github.com/user-attachments/assets/694534a5-6c1b-4397-8b64-0dca0d7b6837" />

**Comment by scraed** [comment]
pls check the latest flux klein workflow, I have added an option for reference

**Comment by houcem-fehri** [comment]
Most of the inconsistencies faded and output is far cleaner than before when switching to t2i mode. Thank you!

Get Outlook for iOS<https://aka.ms/o0ukef>
________________________________
